apps/arena/lib/arena/game_updater.ex

```diff
@@ -674,6 +679,36 @@ defmodule Arena.GameUpdater do
     %{game_state | players: players}
   end
 
+  defp activate_traps(game_state) do
+    now = System.monotonic_time(:millisecond)
+
+    activated_traps =
+      Enum.filter(game_state.traps, fn {_trap_id, trap} ->
+        trap.activate_at < now && trap.aditional_info.status == :PREPARED
+      end)
+
+    {game_state, traps} =
+      Enum.reduce(activated_traps, {game_state, game_state.traps}, fn {trap_id, trap}, {game_state_acc, traps_acc} ->
+        game_state = Trap.do_mechanics(game_state_acc, trap, trap.aditional_info.mechanics)
+        trap = put_in(trap, [:aditional_info, :status], :TRIGGERED)
+        {game_state, Map.put(traps_acc, trap_id, trap)}
```

Comment:
You can reduce the game state directly and use `update_entity_in_game_state` to update the trap, if you decide to do this you should also update `get_entity_path`

Reply:
Great! thanks for the tip